native/Makefile

```diff
@@ -17,18 +17,29 @@ endif
 
 ODIR=obj
 LIBS=-lm
-DEPS = k.h wire.h
+DEPS=k.h wire.h mock_k.h shmipc.h
+OBJECTS=$(ODIR)/shmipc.o $(ODIR)/mock_k.o
 
-all: obj/hpet.so obj/shmipc.so obj/shmmain
+all: $(ODIR)/mock_k.o $(ODIR)/shmipc.o $(ODIR)/shmmain $(ODIR)/libocamlshmipc.so $(ODIR)/ocamlshmipcmain
 
 k.h:
 	wget https://raw.githubusercontent.com/KxSystems/kdb/master/c/c/k.h
 
-$(ODIR)/%.so: %.c $(DEPS)
-	$(CC) -o $@ $< $(CFLAGS) $(CDFLAGS)
 
-$(ODIR)/shmmain: shmmain.c shmipc.c mock_k.h $(DEPS)
-	$(CC) -o $@ $< $(CFLAGS) -g -O0
+$(ODIR)/mock_k.o: mock_k.c $(DEPS)
```

Comment:
mock_k should not be build as a so, it should be compiled in for non-kdb bindings.